src/derivative_operators/convolutions.jl

```diff
@@ -12,14 +12,21 @@ function convolve_interior!(x_temp::AbstractVector{T}, x::AbstractVector{T}, A::
     @assert length(x_temp)+2 == length(x)
     stencil = A.stencil_coefs
     coeff   = A.coefficients
-    mid = div(A.stencil_length,2)
+    # Upwind operators have a non-centred stencil
+    if use_winding(A)
```

Comment:
this is mixing in the winding changes?

Reply:
@ChrisRackauckas Yes.

Reply:
That is confusing the heck out of it. Can you put the changes that are not related in a different branch/PR?

Reply:
sure. Sorry about that.